### DISCUSSION TITLE
Determining which SAFE file contains a coordinate

### DISCUSSION THREAD

**Discussion by ys272** [CS-SI/eodag#432] [Q&A]
Is there an offline way of determining which SAFE file a coordinate belongs to?
I know EODAG provides an API for this purpose, but I have a very large set of coordinates that interest me.

**Comment by sbrunato** [comment]
Hello @ys272 , offline filtering can be performed using eodag crunches, see [API User Guide / Crunch](https://eodag.readthedocs.io/en/stable/notebooks/api_user_guide/6_crunch.html). 
Filtering can be perfomed using a geometry or a property if the coordinate exists in the  EOProducts properties (EOProducts obtained with `search()`). The property can be a tile id if the the provider returns it in the metadata.

For offline matching with MGRS tiles you can also have a look to https://github.com/CS-SI/eotile
